### PR TITLE
Added recipe for pgcgap

### PR DESCRIPTION
**Pull request by liaochenlanruo**
Added recipe for pgcgap

:information_source:
If this is your first pull request to Bioconda please ping `@bioconda/core` so it can be reviewed and to request being added as a member of the Bioconda team. Members of the Bioconda team are able to merge their own pull requests once tests and linting have passed.

----------------

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by liaochenlanruo** [comment]
@bioconda/core
I think the pull request is passing tests, would you like to help me merge the pgcgap?
And I also want to join Bioconda.
Thanks a lot!

**Comment by druvus** [comment]
@liaochenlanruo It seems you missed to add the files

File: recipes/pgcgap/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+source:
+  url: https://github.com/liaochenlanruo/pgcgap/archive/v{{ version }}.tar.gz
+  sha256: '{{ sha256 }}'
+
+build:
+  number: 0
+
+requirements:
+  build:
```
Comment (dpryan79) [inline recipes/pgcgap/meta.yaml]: Can you rename this to `host:`?

File: recipes/pgcgap/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  commands:
+    - pgcgap --check-external-programs
+    - pgcgap --version
+    - pgcgap --help
+
+
+about:
+  home: https://github.com/liaochenlanruo/pgcgap/blob/master/README.md
+  license: GPLv3
+  license_family: GPL
```
Comment (dpryan79) [inline recipes/pgcgap/meta.yaml]: Does this package happen to ship a copy of its license? If so, please add the path to it in `license_file:`

File: recipes/pgcgap/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  summary: A prokaryotic genomics and comparative genomics analysis pipeline
+  dev_url: https://github.com/liaochenlanruo/pgcgap/tree/master
+
+extra:
+  identifiers:
+    - biotools:'{{ name }}'
```
Comment (dpryan79) [inline recipes/pgcgap/meta.yaml]: Please add the following to the `extra:` section:

      recipe-maintainers:
        - liaochenlanruo

We're hoping to better integrate that in the future, so the autobump stuff can ping you and maybe edits to this file can auto-request your review.

**Comment by dpryan79** [comment]
I've reverted the build number bumps for unrelated packages and merged the updated `master` branch back in.

File: recipes/pgcgap/meta.yaml (hunk 1 above)
Comment (liaochenlanruo) [reply recipes/pgcgap/meta.yaml]: Do you mean that I should change "build:" to "host:"?
Thanks for your help!

**Comment by dpryan79** [comment]
I wonder if orthofinder needs a rebuild.

File: recipes/pgcgap/meta.yaml (hunk 1 above)
Comment (dpryan79) [reply recipes/pgcgap/meta.yaml]: Yes

**Comment by liaochenlanruo** [comment]
> I wonder if orthofinder needs a rebuild.

I don't think it need that, thanks!

File: recipes/pgcgap/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+package:
+  name: '{{ name }}'
+  version: '{{ version }}'
+
+source:
+  url: https://github.com/liaochenlanruo/pgcgap/archive/v{{ version }}.tar.gz
+  sha256: '{{ sha256 }}'
+
+build:
+  number: 0
```
Comment (dpryan79) [inline recipes/pgcgap/meta.yaml]: Please add `noarch: generic`.

**Comment by liaochenlanruo** [comment]
@dpryan79 
The test was failed, but I have no idea about how to handle it. Would you like to give me some clues? Thank you very much!
The testing information is in the link below:
https://circleci.com/gh/bioconda/bioconda-recipes/57553#tests/containers/0

**Comment by dpryan79** [comment]
It fails because of the dependency conflict I mentioned before.

**Comment by dpryan79** [comment]
It appears that `gubbins` ends up needing python 3, whereas orthofinder needs python 2.7.

**Comment by liaochenlanruo** [comment]
> It appears that `gubbins` ends up needing python 3, whereas orthofinder needs python 2.7. I don't know enough about either package to know whether those restrictions should actually be correct.
@dpryan79 
Does that mean that in this recipe, we can only keep either gubbins or orthofinder? Or is there a way to keep both dependency packages？

Thank you for your patient guidance！

**Comment by dpryan79** [comment]
Those packages are currently mutually exclusive. I don't know if they really should be or not, someone familiar with them would need to reply.

File: recipes/perl-file-spec/3.48_01/meta.yaml [hunk 5]
```diff
@@ -22,7 +22,10 @@ requirements:
   run:
     - perl
 
-# test:
+  test:
+    commands:
+      - perl -v
+
```
Comment (dpryan79) [inline recipes/perl-file-spec/3.48_01/meta.yaml]: Don't modify random recipes as part of your PR. If linting fails, just:

    git checkout master
    git pull upstream master
    git checkout pgcgap
    git merge master

Comment (liaochenlanruo) [reply recipes/perl-file-spec/3.48_01/meta.yaml]: Ok, sorry about that, thanks!

**Comment by liaochenlanruo** [comment]
Dear @dpryan79 , all  tests were passed, what should I do next?
Thanks!

**Comment by dpryan79** [comment]
I've made a clean version of this in https://github.com/bioconda/bioconda-recipes/pull/15455 Once it's green I'll merge it.

There are too many files deleted from other recipes for me to be able to merge this.

**Pull request closed without merging**